### PR TITLE
[GROW-1661] Launch global nav bar experiment

### PR DESCRIPTION
**Pull request by anandaroop**
https://artsyproduct.atlassian.net/browse/GROW-1661

This sets up the A/B experiment for the desktop global nav bar.

### Control

before

### Experiment 

after

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Danger failed to run `dangerfile.ts`.
  </td>
    </tr>
  </tbody>
</table>



## Error GraphQLError
```
Syntax Error: Cannot parse the unexpected character "<".
GraphQLError: Syntax Error: Cannot parse the unexpected character "<".
    at syntaxError (/home/circleci/project/node_modules/graphql/error/syntaxError.js:24:10)
    at readToken (/home/circleci/project/node_modules/graphql/language/lexer.js:302:32)
    at Object.lookahead (/home/circleci/project/node_modules/graphql/language/lexer.js:61:43)
    at Object.advanceLexer [as advance] (/home/circleci/project/node_modules/graphql/language/lexer.js:52:33)
    at expect (/home/circleci/project/node_modules/graphql/language/parser.js:1296:11)
    at parseDocument (/home/circleci/project/node_modules/graphql/language/parser.js:107:3)
    at parse (/home/circleci/project/node_modules/graphql/language/parser.js:38:10)
    at buildSchema (/home/circleci/project/node_modules/graphql/utilities/buildASTSchema.js:447:43)
    at _callee3$ (/home/circleci/project/scripts/validateSchemas.js:154:31)
    at tryCatch (/home/circleci/project/node_modules/regenerator-runtime/runtime.js:45:40)
```
### Dangerfile
```
19|   const changes: string[] = await getBreakingChanges(versionToCheck, 2)
20| 
21|   // There are breaking changes with the schema
22|   if (changes.length) {
23|     if (isReleasePR) {
------------^
24|       const deployURL = "https://github.com/artsy/metaphysics#deployment"
25|       fail(
26|         `You need to promote Metaphysics from staging to production. The schema used in Reaction, is further ahead than the version in Metaphyisc production. This could cause your queries to fail. You can see how to [deploy here](${deployURL}).`
27|       )
```
  
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d7bf33a3c60656d713d4d4e6ed8249b2da038587
</p>

**Comment by anandaroop** [comment]
@xtina-starr the DNS is still propagating for the review app for this change, but I _expect_ it to show up here shortly: https://force-navbar-test.artsy.net

Please share that with @abhitip when y'all touch base.

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Danger failed to run `dangerfile.ts`.
  </td>
    </tr>
  </tbody>
</table>



## Error GraphQLError
```
Syntax Error: Cannot parse the unexpected character "<".
GraphQLError: Syntax Error: Cannot parse the unexpected character "<".
    at syntaxError (/home/circleci/project/node_modules/graphql/error/syntaxError.js:24:10)
    at readToken (/home/circleci/project/node_modules/graphql/language/lexer.js:302:32)
    at Object.lookahead (/home/circleci/project/node_modules/graphql/language/lexer.js:61:43)
    at Object.advanceLexer [as advance] (/home/circleci/project/node_modules/graphql/language/lexer.js:52:33)
    at expect (/home/circleci/project/node_modules/graphql/language/parser.js:1296:11)
    at parseDocument (/home/circleci/project/node_modules/graphql/language/parser.js:107:3)
    at parse (/home/circleci/project/node_modules/graphql/language/parser.js:38:10)
    at buildSchema (/home/circleci/project/node_modules/graphql/utilities/buildASTSchema.js:447:43)
    at _callee3$ (/home/circleci/project/scripts/validateSchemas.js:154:31)
    at tryCatch (/home/circleci/project/node_modules/regenerator-runtime/runtime.js:45:40)
```
### Dangerfile
```
19|   const changes: string[] = await getBreakingChanges(versionToCheck, 2)
20| 
21|   // There are breaking changes with the schema
22|   if (changes.length) {
23|     if (isReleasePR) {
------------^
24|       const deployURL = "https://github.com/artsy/metaphysics#deployment"
25|       fail(
26|         `You need to promote Metaphysics from staging to production. The schema used in Reaction, is further ahead than the version in Metaphyisc production. This could cause your queries to fail. You can see how to [deploy here](${deployURL}).`
27|       )
```
  
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c812eff0cb3c7f5ab8972de761b7f4db3e72afd4
</p>

File: src/desktop/components/react/stitch_components/index.tsx [hunk 1]
```diff
@@ -1,10 +1,19 @@
 export { SearchBar } from "./SearchBar"
-export { NavBar } from './NavBar'
+export { NavBar } from "./NavBar"
+export { NavBarTest } from "./NavBarTest"
 
-export { ReactionArtworkArtistInfo as ArtworkArtistInfo } from "./ReactionArtworkArtistInfo"
-export { ArtworkDetailsQueryRenderer as ArtworkDetails } from "reaction/Apps/Artwork/Components/ArtworkDetails"
+export {
+  ReactionArtworkArtistInfo as ArtworkArtistInfo,
+} from "./ReactionArtworkArtistInfo"
```
Comment (anandaroop) [inline src/desktop/components/react/stitch_components/index.tsx]: I'm also seeing a lot of whitespace formatting churn, as [others have noted](https://artsy.slack.com/archives/CP9P4KR35/p1576186698265400). Haven't tracked down the source of this 🤔

File: src/desktop/components/main_layout/header/templates/index.jade [hunk 2]
```diff
@@ -1,4 +1,7 @@
 include ../../../util/activator
 
 header
-  != stitch && stitch.components.NavBar({ notificationCount: sd.NOTIFICATION_COUNT, searchQuery: sd.searchQuery, user: sd.CURRENT_USER,  })
+  if sd.DESKTOP_GLOBAL_NAVIGATION == "experiment"
+    != stitch && stitch.components.NavBarTest({ notificationCount: sd.NOTIFICATION_COUNT, searchQuery: sd.searchQuery, user: sd.CURRENT_USER,  })
+  else
+    != stitch && stitch.components.NavBar({ notificationCount: sd.NOTIFICATION_COUNT, searchQuery: sd.searchQuery, user: sd.CURRENT_USER,  })
```
Comment (anandaroop) [inline src/desktop/components/main_layout/header/templates/index.jade]: This is the big conditional at the heart of this PR.

**Comment by anandaroop** [comment]
Since we're holding off on this, I've released the review app and cname

**Comment by anandaroop** [comment]
We have the green light to launch this, looks like the first thing I need to do is rebase.

**Pull request merged**